ui/server/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/DefaultOAuth2ServiceFactory.scala

```diff
@@ -0,0 +1,77 @@
+package pl.touk.nussknacker.ui.security.oauth2
+
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.ui.security.api.GlobalPermission.GlobalPermission
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.ui.security.api.Permission.Permission
+import pl.touk.nussknacker.ui.security.oauth2.DefaultOAuth2ServiceFactory.{OAuth2AuthenticateData, OAuth2Profile}
+import pl.touk.nussknacker.ui.security.oauth2.OAuth2ClientApi.{DefaultAccessTokenResponse, DefaultProfileResponse}
+import pl.touk.nussknacker.ui.security.oauth2.OAuth2ServiceProvider.{OAuth2Service, OAuth2ServiceFactory}
+import sttp.client.{NothingT, SttpBackend}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+
+class DefaultOAuth2Service(clientApi: OAuth2ClientApi[DefaultProfileResponse, DefaultAccessTokenResponse], configuration: OAuth2Configuration) extends OAuth2Service with LazyLogging {
+  override def authenticate(code: String): Future[OAuth2AuthenticateData] = {
+    clientApi.accessTokenRequest(code).map{ resp =>
+      OAuth2AuthenticateData(
+        access_token = resp.access_token,
+        token_type = resp.token_type,
+        refresh_token = resp.refresh_token
+      )
+    }
+  }
+
+  override def profile(token: String): Future[OAuth2Profile] = {
+    clientApi.profileRequest(token).map{ profile =>
+      val roles = DefaultOAuth2ServiceFactory.getUserRoles(profile.email, configuration, List.apply(DefaultOAuth2ServiceFactory.defaultUserRole))
+      OAuth2Profile(
+        id = profile.id.toString,
+        email = profile.email,
+        isAdmin = OAuth2AuthenticatorFactory.isAdmin(roles, configuration.rules),
```

Comment:
If you pass everywhere roles AND rules AND everywhere you do the same filtering, it means that you need to have some smarter classes... :) Why not to replace upper `getUserRoles` invocation with `userMatchingRules`/`userMatchingRoles` or sth similar?

Reply:
Changed.